src/Enhancer/ConditionalEnhancer.php

```diff
@@ -48,12 +61,75 @@ public function __construct(array $enhancerMap)
      */
     public function enhance(array $defaults, Request $request)
     {
-        foreach ($this->enhancerMap as $pair) {
-            if ($pair['matcher']->matches($request)) {
-                return $pair['enhancer']->enhance($defaults, $request);
+        foreach ($this->getEnhancerMap() as $pair) {
+            if (($pair[self::MAP_KEY_MATCHER] instanceof RequestMatcherInterface
```

Comment:
i notice that i got confused (again) by HttpFoundation and Routing both having a RequestMatcherInterface but with different method name... we should decide which of the two makes sense (i think the routing one) and force to only use that. no use in mixing the two kind of request matcher here.